src/library/scala/concurrent/package.scala

```diff
@@ -197,8 +200,12 @@ package concurrent {
     @throws(classOf[TimeoutException])
     @throws(classOf[InterruptedException])
     final def result[T](awaitable: Awaitable[T], atMost: Duration): T = awaitable match {
-      case f: Future[T] if f.isCompleted => f.result(atMost)(AwaitPermission)
+      case CompletedFuture(result: Try[T] @unchecked) => result.get
```

Comment:
Is the type annotation really needed here?
```suggestion
      case CompletedFuture(result) => result.get
```
If `T` cannot be inferred, it's better to use an `asInstanceOf` on the right. It's more honest:
```suggestion
      case CompletedFuture(result) => result.get.asInstanceOf[Try[T]]
```